Creating ImageWithSideTitle

alaasaedy opened the pull request:
- Create mageWithSideTitle Component with storybook

alaasaedy committed:
creating ImageWithSideTitle component

alaasaedy committed:
Creating ImageWithSideTitle

create imagewithsidetitle coponent with story

MAlngaawy commented on src/components/ImageWithSideTitle/index.jsx:
```diff
@@ -0,0 +1,13 @@
+
+const ImageWithSideText = ({ imgUrl, title }) => {
+  return (
+    <div className="flex justify-around bg-c700 rounded py-2 hover:bg-c100 hover:opacity-75">
```
why you used flex here!

- avoid using flex if it's not necessary
- give it specific padding to be like the design
- background and color should back to the component as a prop, and cover all the design colors

MAlngaawy reviewed (changes requested):
nice work Alaa, but still need a lot of changes.
please consider these changes :point_down: 
**and please read the ticket carefully** and cover all the given examples at the ticket description

alaasaedy committed:
- Add Title Color Option to cover all design colors.
- Add condintional subtitle which will render when it has a value as prop.

alaasaedy committed:
Merge branch 'develop' into feature/image-with-side-title

MAlngaawy commented on src/components/ImageWithSideTitle/index.jsx:
```diff
@@ -0,0 +1,49 @@
+const ImageWithSideText = ({ imgUrl, title, titleColor, subTitle }) => {
+  return (
+    <div className="flex justify-around rounded py-2">
+      <img src={imgUrl} alt="" />
```
- should have space between the image and the title ( put the whole div in a button to see what I mean  )
- this image should be an icon that can change its color by a prop 
- and should test it like the post example

**Give me an example by storybook like this format**
![Screenshot from 2020-10-05 18-54-33](https://user-images.githubusercontent.com/46047009/95109107-69d60d00-073c-11eb-9783-31df0124c89e.png)

MAlngaawy reviewed (changes requested):
please consider these changes

alaasaedy committed:
Merge branch 'develop' into feature/image-with-side-title

alaasaedy committed:
- Restructuring the component to fit different cases.
- Add optional background.
- Use of Heading component and adding the required props.

alaasaedy committed:
Merge branch 'feature/image-with-side-title' of https://github.com/m3ntorship/pickly.io into feature/image-with-side-title

alaasaedy committed:
removing unnecessary tag

MAlngaawy reviewed (changes requested):
please, consider these changes
- the space between the title and subtitle here
![Screenshot from 2020-10-06 00-30-52](https://user-images.githubusercontent.com/46047009/95138936-b553df80-076b-11eb-81e7-244d949fa74a.png)

- you covered the background color for the icon, but we don't need it, we need color options for the icon itself, and the title
![Screenshot from 2020-10-06 00-31-30](https://user-images.githubusercontent.com/46047009/95139118-095ec400-076c-11eb-9a32-e21189297110.png)

- I asked you to represent this component by your reusable component.
![Screenshot from 2020-10-05 18-54-33](https://user-images.githubusercontent.com/46047009/95139207-3ad78f80-076c-11eb-856e-7335d13ceccc.png)

- But instead, I found this, and it's not the same
![Screenshot from 2020-10-06 00-39-39](https://user-images.githubusercontent.com/46047009/95139297-71150f00-076c-11eb-9d4e-a8c081987622.png)

- when I tried to pass an avatar image instead of your icon image I found this thing
![Screenshot from 2020-10-06 00-32-16](https://user-images.githubusercontent.com/46047009/95139347-91dd6480-076c-11eb-8114-4ada114fdffe.png)

**Please, reread the last changes  requested, and consider these changes**

**and according to the reusable heading usage**, it has a just one style (14px, 600 font-weight ), you set its props to be like the design, not the one who will use your component

MAlngaawy commented:
and Pleeease, **don't add a label to the ticket**

alaasaedy committed:
- remove bg color object which became unwanted.
- make optional to pass image or icon.
- change img size either it's image or icon.
- passing static props to heading component to match design cases.

MAlngaawy committed:
add comment title color options to be known by the used when use it

MAlngaawy committed:
add catigory components to the component title

MAlngaawy reviewed (approved):
LGTM
